continuous-integration.md

```diff
@@ -7,18 +7,28 @@
 Continuous Integration
 ======================
 
-Continuous Integration (CI) is a practice where developers frequently push their code into a shared repository. 
-Automated builds and checks run on each check-in, allowing teams to detect problems early.
+[Continuous Integration (CI)](https://www.martinfowler.com/articles/continuousIntegration.html) is a practice where developers frequently integrate their work frequently, usually each person integrates at least daily, leading to multiple integrations per day. Each integration is verified by an automated build and checks run on each check-in, allowing teams to detect problems early.
 
 ## Tasks
 
 You should have continuous integration set up in your repository running the following tasks where relevant:
+* Linting your codebase
 * Running unit and integration tests
 * Building your application
-* Linting your codebase
 * Publishing artifacts
 * Other appropriate tasks for your repository
 
+## Keeping the process fast
+
+The purpose of CI is to have rapid feedback and as such you should favour `speedness` over `correctness` and direct  `costs`.    
+Quoting `Martin Folwer`:  
+
+_For most projects, however, the XP guideline of a ten minute build is perfectly within reason. Most of our modern projects achieve this. It's worth putting in concentrated effort to make it happen, because every minute you reduce off the build time is a minute saved for each developer every time they commit. Since CI demands frequent commits, this adds up to a lot of time._
```

Comment:
nitpick on markdown syntax, prefer `>` for quotes:

```suggestion
> For most projects, however, the XP guideline of a ten minute build is perfectly within reason. Most of our modern projects achieve this. It's worth putting in concentrated effort to make it happen, because every minute you reduce off the build time is a minute saved for each developer every time they commit. Since CI demands frequent commits, this adds up to a lot of time.
```